File: Editorial/2094_Finding_3_Digit_Even_Numbers/CSharp/2_Traverse_All_Possible_3_Digit_Even_Numbers.cs
public class Solution {
    public int[] FindEvenNumbers(int[] digits) {
        List<int> res = new List<int>();
        Dictionary<int, int> freq = new Dictionary<int, int>();
        // Count the number of occurrences of each number in the integer array
        foreach (int d in digits) {
            freq[d] = freq.ContainsKey(d) ? freq[d] + 1 : 1;
        }
        // Enumerate all three-digit even numbers
        for (int i = 100; i < 1000; i += 2) {
            Dictionary<int, int> freq1 = new Dictionary<int, int>();
            int num = i;
            // Count the frequency of each digit of the current even number
            while (num > 0) {
                int d = num % 10;
                freq1[d] = freq1.ContainsKey(d) ? freq1[d] + 1 : 1;
                num /= 10;
            }
            // Check if the conditions are met
            bool isValid = true;
            foreach (var entry in freq1) {
                if (!freq.ContainsKey(entry.Key) ||
                    freq[entry.Key] < entry.Value) {
                    isValid = false;
                    break;
                }
            }
            if (isValid) {
                res.Add(i);
            }
        }
        return res.ToArray();
    }
}

File: Editorial/2094_Finding_3_Digit_Even_Numbers/JavaScript/2_Traverse_All_Possible_3_Digit_Even_Numbers.js
var findEvenNumbers = function (digits) {
    const res = [];
    const freq = new Map();
    // Count the number of occurrences of each number in the integer array
    for (const d of digits) {
        freq.set(d, (freq.get(d) || 0) + 1);
    }
    // Enumerate all three-digit even numbers
    for (let i = 100; i < 1000; i += 2) {
        const freq1 = new Map();
        let num = i;
        // Count the frequency of each digit of the current even number
        while (num > 0) {
            const d = num % 10;
            freq1.set(d, (freq1.get(d) || 0) + 1);
            num = Math.floor(num / 10);
        }
        // Check if the conditions are met
        let isValid = true;
        for (const [d, count] of freq1.entries()) {
            if ((freq.get(d) || 0) < count) {
                isValid = false;
                break;
            }
        }
        if (isValid) {
            res.push(i);
        }
    }
    return res;
};
